backend/admission.go

```diff
@@ -0,0 +1,114 @@
+package backend
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+)
+
+// AdmissionHandler manages how objects before they are sent to storage
+type AdmissionHandler interface {
+	ValidateAdmission(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+	MutateAdmission(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+	ConvertObject(context.Context, *ConversionRequest) (*AdmissionResponse, error)
+}
+
+type ValidateAdmissionFunc func(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+type MutateAdmissionFunc func(context.Context, *AdmissionRequest) (*AdmissionResponse, error)
+type ConvertObjectFunc func(context.Context, *ConversionRequest) (*AdmissionResponse, error)
+
+// Operation is the type of resource operation being checked for admission control
+// https://github.com/kubernetes/kubernetes/blob/v1.30.0/pkg/apis/admission/types.go#L158
+type AdmissionRequestOperation int32
+
+const (
+	AdmissionRequestCreate AdmissionRequestOperation = 0
+	AdmissionRequestUpdate AdmissionRequestOperation = 1
+	AdmissionRequestDelete AdmissionRequestOperation = 2
+)
+
+// String textual representation of the operation.
+func (o AdmissionRequestOperation) String() string {
+	return pluginv2.AdmissionRequest_Operation(o).String()
+}
+
+// Identify the Object properties
+type GroupVersionKind struct {
+	Group   string `json:"group,omitempty"`
+	Version string `json:"version,omitempty"`
+	Kind    string `json:"kind,omitempty"`
+}
+
+// AdmissionRequest contains information from a kubernetes Admission request and decoded object(s).
+// See: https://github.com/kubernetes/kubernetes/blob/v1.30.0/pkg/apis/admission/types.go#L41
+// And: https://github.com/grafana/grafana-app-sdk/blob/main/resource/admission.go#L14
```

Comment:
I think it's a bit weird that we are referencing Kubernetes code here in this package. It's okay if we apply the same concepts in this "legacy" package but comments should be self-explanatory, without pointing to code that does not apply to the traditional plugin workflow.